cmd/node/main.go

```diff
@@ -2748,6 +2748,7 @@ func createScQueryElement(
 			BlockGasLimit:                  economics.MaxGasLimitPerBlock(shardCoordinator.SelfId()),
 			GasSchedule:                    gasScheduleNotifier,
 			ArgBlockChainHook:              argsHook,
+			InitialEpoch:                   epochNotifier.CurrentEpoch(),
```

Comment:
no need for this. You can take this out of epochNotifier - as you already send that to the vmContainerFactory as argument. Or you should.

Reply:
Removed `InitialEpoch`, replaced with `EpochNotifier`.